tests/training_success/explore_schema.py

```diff
@@ -0,0 +1,417 @@
+import copy
+import random
+from collections import deque
+from typing import Any, Deque, Dict, List, Tuple, Union
+
+from ludwig.constants import SEQUENCE, TEXT, TIMESERIES
+from ludwig.schema.metadata.parameter_metadata import ExpectedImpact
+from ludwig.schema.model_types.base import ModelConfig
+from ludwig.types import ModelConfigDict
+from ludwig.utils.misc_utils import merge_dict
+
+ParameterBaseTypes = Union[str, float, int, bool, None]
+
+
+def explore_properties(
```

Comment:
@abidwael how difficult do you think it would be to move this from using the jsonschema version of the config schema to using the schema objects like we do in the ludwig-docs: https://github.com/ludwig-ai/ludwig-docs/blob/master/main.py

I say this because I worry that over time this could become very difficult to maintain in its current form, as it's very removed from the representation of the schema that's tied to Ludwig code. What do you think?

Reply:
I chatted with @ksbrar and our conclusion was that 
1. there isn't a streamlined way to access the fields (the equivalent of `properties` dict) at every level of the config. 
```
cls = model_type_schema_registry["ecd"]
defaults = cls.get_class_schema()().fields["defaults"]
text = defaults.cls.get_class_schema()().fields["text"]

text.fields # fails: no attribute fields
text.cls # fails: no attribute cls
```
2. Even if that wasn't a problem, it doesn't look like there is a way to retrieve fields like minimum, maximum, string options that help us infer the values we suggest for the combinatorial configs. Example of what's returned in `stringOptions`:
```
    return field(
        metadata={
            "marshmallow_field": fields.String(
                validate=validate.OneOf(options),
                allow_none=allow_none,
                load_default=default,
                dump_default=default,
                metadata={
                    "description": description,
                    "parameter_metadata": convert_metadata_to_json(parameter_metadata) if parameter_metadata else None,
                },
            )
        },
        default=default,
    )
```
As you can see, the only available options are defaults. We can add more fields to it, but I don't think we should as it will make the schema bigger than it already is and `ludwig.schema.utils.unload_jsonschema_from_marshmallow_class` already does that.

The benefit of `get_schema` is that it also unpacks the `Validator` class of every field which will contain the rules (min, max, options, etc.). These don't exist in the classes themselves and if we want to make this work with objects, we will have to access the validator attributes. Not all validators are native marshmallow validators, though. We define some of them. So if we wanted to implement them the object-oriented way, we'll also need to have logic that checks whether they're marshmallow-native or defined in Ludwig and then get the validator-specific fields (e.g. validator.choices for oneOf validator.)

Therefore, I think the current approach is the right one. I'm not 100% sure I understood your comment about maintainability correctly. If someone wanted to maintain this, they'd just have to understand what's in the JSON schema. If there's documentation work that I can do to improve this, I'm happy to add that.